src/core/qgsmaplayerregistry.cpp

```diff
@@ -158,7 +163,7 @@ void QgsMapLayerRegistry::removeMapLayers( const QList<QgsMapLayer*>& layers )
     emit layerRemoved( myId );
   }
 
-  emit layersRemoved( layerIds );
+  emit layersRemoved( layerList );
```

Comment:
I'm not sure that's correct, last time I looked the `layersRemoved` signal was sending passing ids (and not pointers to possibly deleted layers).